ancestry.go

```diff
@@ -211,9 +211,20 @@ func (b *AncestryBuilder) createLayerIndexedFeature(namespace *layerIndexedNames
 
 func (b *AncestryBuilder) lookupNamespace(feature *database.LayerFeature) (*layerIndexedNamespace, bool) {
 	matchedNamespaces := []*layerIndexedNamespace{}
-	for i, namespace := range b.namespaces {
-		if namespace.Namespace.VersionFormat == feature.VersionFormat {
-			matchedNamespaces = append(matchedNamespaces, &b.namespaces[i])
+	if feature.PotentialNamespace.Name != "" {
+		a := &layerIndexedNamespace{
+			Namespace: database.LayerNamespace{
+				Namespace: feature.PotentialNamespace,
+			},
+			IntroducedIn: b.layerIndex,
+		}
+		matchedNamespaces = append(matchedNamespaces, a)
```

Comment:
We should always ensure that all elements in matched namespaces are in the ancestrybuilder.namespaces, and any feature with the same namespace must point to the same namespace object in the ancestrybuilder.namespaces slice. 

Otherwise, this will break the updateNamespace logic. 

BTW, in order to use the CPE string as a namespace, please remove the panic condition for updateNamespace logic.

Reply:
^^^ 
This may not be the case.
I think for now at least: when a feature has a potential namespace, we will use that namespace directly instead of getting into the namespace lookup and upgrade process. So that the output AncestryFeature is guaranteed to have the namespace.

@jzelinskie What do you think about this logic here?

Reply:
I think this logic is fine for now. I don't want to put the cart before the horse. Let's get things working and then test our assumptions later.